datacollection/nasa-astm-utm.yaml

```diff
@@ -136,6 +136,10 @@ components:
           type: string
           description: |-
             RFC3339-formatted time/date string.  The time zone must be 'Z'.
+            For NASA Data Pipeline date-time is validated against this ECMA 262 regex:
```

Comment:
We want to allow the time/date string to accept four flavors of timestamps (i.e., no ms, 1 digit ms, 2 digit ms, 3 digit ms). Here are 4 examples representing each flavor:

2020-08-27T23:18:39.001Z
2020-08-27T23:18:39.01Z
2020-08-27T23:18:39.1Z
2020-08-27T23:18:39Z

These changes to the Time definition allow openapi codegen to create correct Java code. The above 4 cases are allowed, but other variations are rejected:

          format: "iso8601"
          pattern: "^([0-9]{4})-(1[0-2]|0[1-9])-(3[01]|0[1-9]|[12][0-9])T(2[0-3]|[01][0-9]):([0-5][0-9]):([0-5][0-9])(\\.[0-9]{1,3})?Z$"
          example: "2015-08-20T14:11:56.118Z"

Reply:
we need to (MUST) minimize the changes to ASTM spec.  Without any changes, ASTM spec (Time model) will meet Mark's test cases.  

However the regex on the DMP models (Timestamp model) will need to get this regex.  When x3 is changed both  are same wrt timeformat validation.https://raw.githubusercontent.com/nasa/uam-apis/master/datacollection/openapi/X3/uam-data-collection-X3.yaml

Reply:
If the regular expression in line 141 is adopted, it requires exactly 3 digits for milliseconds. The "example" line shows a time with 2 digits for milliseconds (i.e., 1985-04-12T23:20:50.52Z). This example would fail validation using the regular expression in line 141. This regular expression was added to the description section, not as a pattern value, so it would not be used by any codegen tools.

This is the regular expression that UDC is planning to use to validate timestamps on inbound operations, constraints, telemetry and reports:
pattern: "^([0-9]{4})-(1[0-2]|0[1-9])-(3[01]|0[1-9]|[12][0-9])T(2[0-3]|[01][0-9]):([0-5][0-9]):([0-5][0-9])(\\.[0-9]{1,3})?Z$"

The regular expression is part of the "pattern" line and not the "description" line, so codegen will enforce these exact 4 flavors of date/time strings:

2020-08-27T23:18:39.001Z
2020-08-27T23:18:39.01Z
2020-08-27T23:18:39.1Z
2020-08-27T23:18:39Z

Any other format, including a format currently allowed by the official ASTM v0.3.5 spec will return a HTTP 400 Bad Request.

Reply:
oops we have always been agreeing, but I mistakenly answered this q based on memory that this line of code was never changed.  Pardon.  
Pushing to agree with you, Mark.